_posts/2017-10-11-Stairway-To-Health.md

```diff
@@ -0,0 +1,356 @@
+*timespan.route.ts*
+```typescript
+router.get('/timespan/', (req: Request, res: Response, next: NextFunction) => {
+    this.timespanController.getTimespanList(req, res, next);
+});
+router.post(('/timespan/', this.authenticate, (req: Request, res: Response, next: NextFunction) => {
+    this.timespanController.createTimespan(req, res, next);
+});
+```
+
+next under our `/controllers` directory we have our controllers where all our functionality/logic is<br>
+*timespan.controller.ts*
+```typescript
+public getTimespanList(req: Request, res: Response, next: NextFunction) {
+    return this.timespanModel.find({}, [], {sort: {start: 1}})
+    .then((result) => {
+        res.json(result).status(200);
+    }, (err)=>{
+        res.statusCode = 500;
+        res.statusMessage = err;
+        res.send();
+    });
+}
+```
+
+## Authentication
+To prevent everyone from changing these configurations of course we had to add some authentication functionality. 
+As you can see in the router code above, we created an authentication middleware so that on every route that we want the user to be authenticated, we can simply add `this.authenticate()` to the route. 
+This checks a JWT token in the headers. 
+We check the token to be valid. 
+If it's not valid, we send an `unauthorized` response, and if it is valid, we decode it and add it as a user object on the request. 
+This way we can access it in the controller and do some logic depending on its role, etc.
+`this.authenticate` is a method we added to the `core.route.ts` every route extends this super class so that we can put common code and middleware in this file.
+
+JWT stands for JSON Web Token and is a JSON-based open standard for creating access tokens that assert some number of claims. 
+For example, a server could generate a token that has the claim `logged in as admin` and provide that to a client. 
+The client could then use that token to prove that he is logged in as admin.
+
+## Deploy
+Finally we deployed it to the Proximus data center and watched the Proximus employees take on the challenge.
+
+<div style="text-align: center; margin: 0px auto;">
+    <a href="{{ '/img/stairwaytohealth/result1.jpg' | prepend: site.baseurl }}" data-lightbox="results" data-title="Large screen @ Proximus towers">
+        <img alt="result1" src="{{ '/img/stairwaytohealth/result1.jpg' | prepend: site.baseurl }}" class="image fit" style="width: 61.45%; display: inline-block;">
+    </a>
+    <a href="{{ '/img/stairwaytohealth/result2.jpg' | prepend: site.baseurl }}" data-lightbox="results" data-title="Informing the employees">
+        <img alt="result2" src="{{ '/img/stairwaytohealth/result2.jpg' | prepend: site.baseurl }}" class="image fit" style="width: 34.55%; display: inline-block;">
+    </a>
+</div>
+
+
+## Conclusion
+After four hard weeks of working and a lot of lines of code written. 
```

Comment:
After four hard weeks of working and writing many lines of code, we delivered our project to Proximus and the contest could start.